pgmpy/tests/test_models/test_BayesianNetwork.py

```diff
@@ -1807,23 +1805,77 @@ def test_missing_scheme_sample(self):
         samples = self.con_model.simulate(n_samples=1000)
         self.assertFalse(samples.isnull().values.any())
 
-        samples = self.con_model.simulate(
-            n_samples=1000, include_missing=True, missing_prob=0.9
+        cpd = TabularCPD(
+            "Z*",
+            2,
+            [[0.5], [0.5]],
+        )
+        samples = self.con_model.simulate(n_samples=1000, missing_prob=cpd)
+        self.assertTrue(samples["Z"].isnull().values.any())
+        self.assertFalse(samples.drop(columns=["Z"]).isnull().values.any())
```

Comment:
In the tests, I would suggest testing for the number of missing values to make sure that the method is working as expected. As this is the MCAR case with 0.5 probability, we would expect around 50% of the values to be missing. We can test that by, for example, testing that the fraction of missing values are between 0.45 to 0.55.